tests/e2e/scale/test_cephfs_many_files.py

```diff
@@ -0,0 +1,238 @@
+    Tar that directory to a zipped tar file.
+    Rsynch that tar file to the cephfs pod
+    Extract the tar files on ceph pod onto the mounted ceph filesystem.
+
+    Returns: list of ten of the files created.
+    """
+    if os.path.isfile(JSON_FNAME):
+        os.remove(JSON_FNAME)
+    return_list = []
+    logging.info(f"Creating {TFILES} files on Cephfs")
+    onetenth = TFILES / 10
+    endoften = onetenth - 1
+    ntar_loc = mkdtemp()
+    tarfile = os.path.join(ntar_loc, TARFILE)
+    new_dir = mkdtemp()
+    for i in range(0, TFILES):
+        fname = mktemp(dir=new_dir)
+        with open(fname, 'w') as out_file:
+            out_file.write(SAMPLE_TEXT)
+        if i % onetenth == endoften:
+            dispv = i + 1
+            logging.info(f'{dispv} local files created')
+            return_list.append(fname)
+    tmploc = ntar_loc.split('/')[-1]
+    subprocess.run([
+        'tar',
+        'cfz',
+        tarfile,
+        '-C',
+        new_dir,
+        '.'
+    ])
+    subprocess.run([
+        'oc',
+        '-n',
+        DEFAULT_NS,
+        'rsync',
+        ntar_loc,
+        f'{POD_NAME}:{TARGET_DIR}'
+    ])
+    subprocess.run([
+        'oc',
+        '-n',
+        DEFAULT_NS,
+        'exec',
+        POD_NAME,
+        '--',
+        'mkdir',
+        f'{TARGET_DIR}/x'
+    ])
+    subprocess.run([
+        'oc',
+        '-n',
+        DEFAULT_NS,
+        'exec',
+        POD_NAME,
+        '--',
+        'tar',
+        'xf',
+        f'{TARGET_DIR}/{tmploc}/{TARFILE}',
+        '-C',
+        f'{TARGET_DIR}/x'
+    ])
+    rmtree(new_dir)
+    os.remove(tarfile)
+    return return_list
+
+
+class MillionFilesOnCephfs(object):
+    """
+    Create pvc and cephfs pod, make sure that the pod is running.
+    """
+    def __init__(self):
+        self.cephfs_pvc = helpers.create_pvc(
+            constants.DEFAULT_STORAGECLASS_CEPHFS,
+            pvc_name=PVC_NAME,
+            namespace=DEFAULT_NS,
+            size=SIZE
+        )
+        self.cephfs_pod = helpers.create_pod(
+            interface_type=constants.CEPHFILESYSTEM,
+            pvc_name=self.cephfs_pvc.name,
+            namespace=DEFAULT_NS,
+            node_name='compute-0',
+            pod_name=POD_NAME
+        )
+        helpers.wait_for_resource_state(self.cephfs_pod, "Running", timeout=300)
+        logging.info("pvc and cephfs pod created")
+
+    def cleanup(self):
+        self.cephfs_pod.delete()
+        self.cephfs_pvc.delete()
+        logging.info("Teardown complete")
+
+
+@pytest.fixture(scope='session')
+def million_file_cephfs(request):
+    million_file_cephfs = MillionFilesOnCephfs()
+
+    def teardown():
+        million_file_cephfs.cleanup()
+    request.addfinalizer(teardown)
+
+
+@scale
+@ignore_leftovers
+@pytest.mark.parametrize(
+    argnames=["resource_to_delete", "build_temp"],
+    argvalues=[
+        pytest.param(
+            *['mgr', True],
+        ),
+        pytest.param(
```

Comment:
Since this test is for 1 million files, let's create 1 million files for all the tests and not just `mgr`. So, maybe you can remove this param and create 1 million files for all the tests. Or, 1 million files could be created as part of the setup.

Reply:
I had previously ended up creating the files each time this ran.  Thast second parameter is probably not the right soluion.  I will try to reengineer this to create the 1,000,000 files once, and earlier.